### PR TITLE
fix(deploy): correct sha and environment detection 

### LINKED ISSUE #3315: Automatic deployment to staging/prod not working
The deployment workflow fails when triggered automatically via repository_dispatch event:
https://github.com/ecamp/ecamp3/actions/runs/4302780006/jobs/7501807543

Suceeeds when triggered manually via workflow_dispatch event:
https://github.com/ecamp/ecamp3/actions/runs/4302835287/jobs/7501829261

Seems `context.payload.ref` is not set in the former case. I guess we could just use `github.event.client_payload.ref` as in the if-clause.

### PR DESCRIPTION
**Pull request by usu**
Fixes #3315

- Fixes workflow failure, when deployment is triggered automatically by ci-passed event
- Also fixes wrong SHA being deployed (so far, the workflow would have always deployed the SHA from devel for automatic deployments)

Tested as good as possible on my own repo

### CODE REVIEW

File: .github/workflows/deployment.yml [hunk 1]
```diff
@@ -28,11 +24,14 @@ jobs:
         with:
           result-encoding: string
           script: |
-            return context.payload.ref.replace('refs/heads/', '')
-  
+            return (context.payload.action === 'ci-passed' ? context.payload.client_payload.ref : context.payload.ref).replace('refs/heads/', '')
```
Comment (BacLuc) [inline .github/workflows/deployment.yml:27]: a script may have more than one line :wink:

File: .github/workflows/deployment.yml [hunk 2]
```diff
@@ -42,49 +41,17 @@ jobs:
...
-          echo "Git SHA: $REF"
-          echo "sha=$REF" | tr -d "\n" >> $GITHUB_OUTPUT
-
-          echo "github:"
-          <<-HEREDOC
-            ${{ toJSON(github) }}
-          HEREDOC
-          echo ""
-          echo "inputs:"
-          <<-HEREDOC
```
Comment (BacLuc) [inline .github/workflows/deployment.yml:60]: I will add this again when i make the workflow repository independent.
Its very useful to detect newlines in environment variables and secrets which break the deployment

Comment (usu) [reply .github/workflows/deployment.yml:60]: For me it didn't echo anything, so I might have used it wrongly

**Review by carlobeltrame — APPROVED** [review]
LGTM

File: .github/workflows/deployment.yml (hunk 2 above)
Comment (BacLuc) [reply .github/workflows/deployment.yml:60]: no, the cat was forgotten on most debug outputs. as i said, i will add it again, no problem

**Review by BacLuc — APPROVED** [review]

File: .github/workflows/deployment.yml (hunk 2 above)
Comment (usu) [reply .github/workflows/deployment.yml:60]: Ah I see, thanks

**Pull request merged**